src/compositions/useWalletConnectV2.ts

```diff
@@ -0,0 +1,161 @@
+import EthereumProvider from '@walletconnect/ethereum-provider'
+import { setState, useState, getState } from '@/stores'
+import useWeb3 from '@/compositions/useWeb3'
+import { provider as Provider } from 'web3-core'
+import {
+  DEFAULT_NETWORK_CONFIG,
+  WALLET_CONNECT_PROJECT_ID,
+} from '@/helpers/config'
+
+let provider: EthereumProvider
+
+/**
+ * Prepares a WalletConnect V2 provider (EthereumProvider) that manages the WebSocket connection.
+ */
+const setupWCV2Provider = async (): Promise<void> => {
+  const { setupWeb3 } = useWeb3()
+
+  provider = await EthereumProvider.init({
+    projectId: WALLET_CONNECT_PROJECT_ID,
+    chains: [DEFAULT_NETWORK_CONFIG.chainId],
+    methods: [
+      'eth_sendTransaction',
+      'eth_sign',
+      'personal_sign',
+      'eth_getBalance',
+      'eth_getAccounts',
+      'eth_requestAccounts',
+      'up_import',
+      'up_addTransactionRelayer',
+    ],
+    metadata: {
+      name: 'UP Test DApp',
+      description: 'UP Test DApp',
+      url: 'https://up-test-dapp.lukso.tech/',
```

Comment:
Wondering if we can get the URL of the DApp at runtime (maybe there's a value in some environment variables)?

This way we won't have to change the value here in case the URL will change later.
Also, the extension is probably comparing this URL with the URL of the web page the request came from (probably, I'm note sure).

Reply:
The URL is `document.location.origin`